### ISSUE TITLE
rclone config does not use --password-command when encrypting configuration

### ISSUE THREAD

**Issue by dzoep** [rclone/rclone#7859]
#### The associated forum post URL from `https://forum.rclone.org`

None.

#### What is the problem you are having with rclone?

Using `rclone config --password-command some-command` does not affect the initial encryption of rclone configuration performed from the `rclone config` menu under "s) Set configuration password".

The `changeConfigPassword` function in `fs/config/crypt.go` uses `ChangePassword` unconditionally, which in turn uses `GetPassword` unconditionally, which uses `ReadPassword` unconditionally - which uses either standard input or direct terminal I/O.

However the `Decrypt` function in `crypt.go` honors the `--password-command` (the `ci.PasswordCommand` obtained from configuration being used) and properly allows for using external programs to get the password value.

This can be used for offloading the password storage/generation into external hardware key or other protected password storage.

Currently, it is necessary to initially encrypt the configuration using direct terminal I/O or standard input only.

#### What is your rclone version (output from `rclone version`)

rclone v1.60.1-DEV

The same behavior can be seen on latest HEAD.

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)

Linux x86_64 (Ubuntu 22.04, 23.10 and 24.04, Debian 11 and 12).

#### Which cloud storage system are you using? (e.g. Google Drive)

Mostly related to the protecting the crypt remote key material. For completeness: the backing cloud storage is S3 backed by Ceph.

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)

`rclone config --password-command 'zenity --password'`

This is NOT a suggested usage, however it shows the problematic behavior nicely.

#### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)

```
2024/05/20 15:23:44 DEBUG : rclone: Version "v1.60.1-DEV" starting with parameters ["rclone" "-vv" "config" "--config" "configs/cesnet-clean.conf" "--password-command" "zenity --password"]
2024/05/20 15:23:44 DEBUG : Using config file from "/home/joe/Trustica/git/cesnet-crypt/configs/cesnet-clean.conf"
Current remotes:

Name                 Type
====                 ====
cesnet-clean         s3
cesnet-encrypted     crypt

e) Edit existing remote
n) New remote
d) Delete remote
r) Rename remote
c) Copy remote
s) Set configuration password
q) Quit config
e/n/d/r/c/s/q> 
```

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by ncw** [comment]
Hi @dzoep I've had a first go at this. The --password-command is now used when setting and changing the config password.

It occurs to me that there is no programmatic way to encrypt the config file other than by going through `rclone config` so maybe we should create that - this would make scripting easier - what do you think?

Please give this a go.

[v1.68.0-beta.8123.46b0c7f16.fix-7859-password-command](https://beta.rclone.org/branch/fix-7859-password-command/v1.68.0-beta.8123.46b0c7f16.fix-7859-password-command/) on branch [fix-7859-password-command](https://github.com/rclone/rclone/tree/fix-7859-password-command) (uploaded in 15-30 mins)

**Comment by ncw** [comment]
I've implemented command line options to mananage the config file encryption

What do you think of these @dzoep ?

[v1.68.0-beta.8262.b0e17c6f8.fix-7859-password-command](https://beta.rclone.org/branch/fix-7859-password-command/v1.68.0-beta.8262.b0e17c6f8.fix-7859-password-command/) on branch [fix-7859-password-command](https://github.com/rclone/rclone/tree/fix-7859-password-command) (uploaded in 15-30 mins)

## rclone config encryption set

This command sets or changes the config file encryption password.

If there was no config password set then it sets a new one, otherwise
it changes the existing config password.

Note that if you are changing an encryption password using
`--password-command` then this will be called once to decrypt the
config using the old password and then again to read the new
password to re-encrypt the config.

When `--password-command` is called to change the password then the
environment variable `RCLONE_PASSWORD_CHANGE=1` will be set. So if
changing passwords programatically you can use the environment
variable to distinguish which password you must supply.

Alternatively you can remove the password first (with `rclone config
encryption remove`), then set it again with this command which may be
easier if you don't mind the unecrypted config file being on the disk
briefly.

```
Usage:
  rclone config encryption set [flags]

Flags:
  -h, --help   help for set
```

## rclone config encryption remove

Remove the config file encryption password

This removes the config file encryption, returning it to un-encrypted.

If `--password-command` is in use, this will be called to supply the old config
password.

If the config was not encrypted then no error will be returned and
this command will do nothing.

```
Usage:
  rclone config encryption remove [flags]

Flags:
  -h, --help   help for remove
```

## rclone config encryption check

This checks the config file is encrypted and that you can decrypt it.

It will attempt to decrypt the config using the password you supply.

If decryption fails it will return a non-zero exit code if using
`--password-command`, otherwise it will prompt again for the password.

If the config file is not encrypted it will return a non zero exit code.

```
Usage:
  rclone config encryption check [flags]

Flags:
  -h, --help   help for check
```

**Comment by ncw** [comment]
Thank you for testing :-)

I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org/) in 15-30 minutes and released in v1.68
